File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode temp = new ListNode(0, head);
        ListNode newNode = temp;

        while(newNode != null){
            while(newNode.next != null && newNode.next.val == val){
                newNode.next = newNode.next.next;
            }
            newNode = newNode.next;
        }
        return temp.next;
    }
}

File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        newNode = ListNode(0,head)
        temp = newNode

        while temp:
            while temp.next and temp.next.val == val:
                temp.next = temp.next.next
            temp = temp.next
        return newNode.next
            
